modules/internal/ChapelBase.chpl

```diff
@@ -720,6 +720,12 @@ module ChapelBase {
     // stone, we could do parallel init for plain old data (POD) types.
     //
 
+    // for uints we need to check that s > 0, so the `s-1` in the following
+    // loops doesn't overflow.
+    if isUint(s) && s == 0 {
+      return;
```

Comment:
I think it's fine as is, given that in the case of ints it will still finish quickly.